### PR TITLE
fix(button): add muitheme to button; add Error button; add reactestlib; fix all tests

### LINKED ISSUE #16: Button: Finish docs for Button

### LINKED ISSUE #19: Tests: Remove enzyme; Use React Testing Library instead. Get theme in there too.
- remove enzyme
- use react testing lib
- add muithemeprovider to rest testing lib render method
- update all tests: ConfirmDialog, DialogTitle, Button, Chip.

### LINKED ISSUE #29: Button: Add theme to Button + AND get the test for that to work.
It's not working.

### PR DESCRIPTION
**Pull request by machikoyasuda**
Signed-off-by: Machiko Yasuda <machiko@reactioncommerce.com>

Resolves #29 - themes
Resolves #16 - docs
Resolves #19 - tests
Impact: **major**  
Type: **bugfix|style|docs|tests**

### Component: Button
- add mui theme to Button
- renames danger button to Error button

### Tests
- remove Enzyme, React Test Renderer
- add https://testing-library.com/docs/react-testing-library, and Jest-dom
- create `renderWithProviders` to add mui theme provider to test renderer

### Packages
- adds `react-testing-library`
- adds `jest-dom`
- removes `enzyme`
- removes `react-test-renderer`

### Templates
- Updates test template to use react-testing-lib and jest-dom.

## Breaking changes
- Danger button is now Error button. Change all `color="danger"` to `color="error"`

## Testing/Reviewing
1. Button docs: https://deploy-preview-32--heuristic-perlman-0ef024.netlify.com/#/Base%20Components/Actions/Button
1. Test ConfirmDialog test
1. Test the tests! Make sure the snapshots reflect reality.
1. Make sure I did not mess up the Babel configs.
1. Make sure the component templates test looks okay.

### CODE REVIEW

**Comment by machikoyasuda** [comment]
What do you think so far of this @rymorgan https://deploy-preview-32--heuristic-perlman-0ef024.netlify.com/#/Base%20Components/Actions/Button ?

**Comment by machikoyasuda** [comment]
@nnnnat I am getting this error when I try running this spec:

```
 FAIL  package/src/components/Button/Button.test.js
  ● Test suite failed to run

    Cannot find module 'core-js/modules/es6.promise' from 'polyfills.js'

      1 | 'use strict';
      2 | 
    > 3 | if (typeof Promise === 'undefined') {
        | ^
      4 |   // Rejection tracking prevents a common issue where React gets into an
      5 |   // inconsistent state due to an error, but it gets swallowed by a Promise,
      6 |   // and the user has no idea what causes React's erratic future behavior.

      at Resolver.resolveModule (node_modules/jest-resolve/build/index.js:230:17)
      at Object.<anonymous> (config/polyfills.js:3:1)

Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total
Time:        0.043s
Ran all test suites matching /package\/src\/components\/Button\/Button.test.js/i.

Watch Usage: Press w to show more.
```

What babel config am I missing?

**Comment by machikoyasuda** [comment]
@nnnnat @mikemurray 

I _think_ I have testing-library/react and the new themeprovider working right. But now I'm still getting the same error I was getting before w/ Enzyme:

``` FAIL  package/src/components/Button/Button.test.js
  ✕ basic snapshot - only default props (35ms)

  ● basic snapshot - only default props

    TypeError: Cannot read property 'red' of undefined

      11 |   containedPrimary: {
      12 |     "color": theme.palette.primary.contrastText,
    > 13 |     "backgroundColor": theme.palette.colors.red,
         |                                             ^
      14 |     "&:hover": {
      15 |       "backgroundColor": theme.palette.colors.redHover,
      16 |       // Reset on touch devices, it doesn't add specificity
```

When I console.log(theme.palette.colors.red) in `tests/index.js`, it returns the red hex value. So the defaultTheme does have this value from `makeStyles((theme)` in Button.js, but still can't seem to render it all in the tests. Any ideas?

**Comment by machikoyasuda** [comment]
Okay this component is ready for a code review!

@mikemurray - Can you test that the Button component code looks good? Esp. the way the theming is handled, the refs are forwarded. Also, check out how I changed the ConfirmDialog test to actually click the button. Would you add any other assert/expect calls?
@nnnnat - I removed enzyme/react-test-renderer from this app, and added react-test-lib and jest-dom instead. Can you review how I made the TestProvider Wrapper and edited some of the babel/config stuff? Anything you'd change on the way the test is configured?

File: package/src/components/ConfirmDialog/ConfirmDialog.test.js [hunk 1]
```diff
@@ -1,20 +1,27 @@
...
+import { render, fireEvent, cleanup } from "../../tests/";
+import "@testing-library/jest-dom/extend-expect";
 import Button from "../Button";
 import ConfirmDialog from "./ConfirmDialog";
 
-test("basic snapshot", () => {
-  const component = renderer.create((
+afterEach(cleanup);
+
+test("basic snapshot - with opening the dialog", () => {
```
Comment (nnnnat) [inline package/src/components/ConfirmDialog/ConfirmDialog.test.js]: I'm not sure how this components works if it gets added to the DOM or if it's just hidden on the page. If it's getting attached to the DOM on open I think you could use [toBeInTheDocument](https://github.com/testing-library/jest-dom#tobeinthedocument) assertion in this test.

File: package/src/components/Button/Button.test.js [hunk 2]
```diff
@@ -1,12 +1,25 @@
...
+  const { asFragment } = render(<Button className="myBtn" color="error" variant="contained">Delete</Button>);
+  expect(asFragment()).toMatchSnapshot();
+});
+
+test("error button snapshot", () => {
+  const { asFragment } = render(<Button className="myBtn" color="error" variant="outlined">Delete</Button>);
+  expect(asFragment()).toMatchSnapshot();
+});
+
+test("isWaiting button snapshot", () => {
```
Comment (nnnnat) [inline package/src/components/Button/Button.test.js]: I believe you could also do a jest-dom [toBeDisabled](https://github.com/testing-library/jest-dom#tobedisabled) assertion here.

File: package.json [hunk 3]
```diff
@@ -75,9 +80,6 @@
     "setupFiles": [
       "<rootDir>/config/polyfills.js"
     ],
-    "setupFilesAfterEnv": [
```
[the remark this answers is not in the document]
Comment (machikoyasuda) [reply package.json]: I changed this by undeleting the `src/testSetup.js` file and importing the 2 modules into there. Just using the `setupFilesAfterEnv` shorthand with the 2 module names didn't work for some reason. But now my tests are failing. Not sure why!

[the remark this answers is not in the document]
Comment (machikoyasuda) [reply package.json]: Fixed!

**Pull request merged**